### PR TITLE
Fixes for 1.21 e2e tests

### PR DESCRIPTION
**Pull request by olemarkus**
* Set the correct jwks path. Should fix the failing OIDC test.
* Feature gate for CSI deregistration changed name. See #10777

### CODE REVIEW

File: pkg/model/components/discovery.go [hunk 1]
```diff
@@ -55,20 +54,8 @@ func (b *DiscoveryOptionsBuilder) BuildOptions(o interface{}) error {
 		kubeAPIServer.ServiceAccountIssuer = &serviceAccountIssuer
 	}
 
-	// We set apiserver ServiceAccountKey and ServiceAccountSigningKeyFile in nodeup
-
-	if useJWKS {
-		if kubeAPIServer.FeatureGates == nil {
-			kubeAPIServer.FeatureGates = make(map[string]string)
-		}
-		kubeAPIServer.FeatureGates["ServiceAccountIssuerDiscovery"] = "true"
```
Comment (hakman) [inline pkg/model/components/discovery.go:64]: This part is removed because `ServiceAccountIssuerDiscovery` is enable by default in 1.20+, right?

Comment (olemarkus) [reply pkg/model/components/discovery.go:64]: Correct

**Comment by hakman** [comment]
👍 
/lgtm

**Pull request merged**